pkg/operator/csidriveroperator/driverstarter.go

```diff
@@ -102,10 +113,35 @@ func (c *CSIDriverStarterController) sync(ctx context.Context, syncCtx factory.S
 	}
 
 	for i := range c.controllers {
-		if c.controllers[i].operatorConfig.Platform == platform && !c.controllers[i].running {
-			klog.V(2).Infof("Starting CSI driver controller manager %s", c.controllers[i].operatorConfig.ConditionPrefix)
-			go c.controllers[i].mgr.Start(ctx)
-			c.controllers[i].running = true
+		ctrl := &c.controllers[i]
+		if ctrl.operatorConfig.Platform != platform {
+			continue
+		}
+		klog.V(4).Infof("CSIDriverStarterController syncing %s", ctrl.operatorConfig.ConditionPrefix)
+
+		if ctrl.olmRemovalController != nil {
+			if !ctrl.olmRemovalControllerRunning {
+				// Start the OLM removal controller
+				klog.V(2).Infof("Starting OLM removal controller for %s", ctrl.operatorConfig.ConditionPrefix)
```

Comment:
nit: v(4)?

Reply:
It should be printed only once (`!ctrl.olmRemovalControllerRunning`) and I'd prefer to know what has been started.